### DISCUSSION TITLE
Command line vs through neovim lsp

### DISCUSSION THREAD

**Discussion by gwww** [microsoft/pyright#3311] [General]
I have a simple little piece of code:
```
def foo(_unused_variable):
    pass
```

When I edit in neovim I get `_used_variable is not accessed`

When I run `pyright foo.py` is comes up clean (`0 errors, 0 warnings, 0 informations`). (BTW, should `informations` be `informationals`?)

My goal, which I have not figured out, is how to turn off the message in neovim. I'm not clear of the state of the pyright codebase with respect to variables starting with `_`. I've read many bug reports but not sure where it stands. I've read the config wiki page too, but am missing something.

Edit: I care about this for parameters only.

Edit 2: I have a `[tool.pyright]` section in my `pyproject.py`. I have tried `reportUnusedVariable` as both `true` and `false`. Neither way changes what happens.

**Comment by erictraut** [comment]
This has been discussed previously. This is a problem in neovim and the LSP plugin that hosts pyright.

The language server protocol distinguishes between "diagnostics" (errors, warning, informational) and "hints", which are similar to diagnostics but are meant to be used somewhat differently. Hints also have specific "tags" that indicate how the hint is meant to be interpreted. One such tag is called "UnusedCode", and clients that support this hint are meant to display the text range as "grayed out" or some similar manner as a subtle visual cue that the code is unused or a symbol is unreferenced. VS Code interprets this hint as intended.

The language server protocol allows a client to indicate which hint tags it supports. Neovim's LSP plugin is telling pyright that it wants to receive the UnusedCode hint, but then neovim displays this hint as a diagnostic.

When you run the command-line version of pyright, it outputs only diagnostics. It doesn't output any hints because those are meant specifically for language server clients and don't have meaning to a command-line tool. 

I think your options are:
1. Change the neovim LSP plugin to indicate that it doesn't support the UnusedCode hint, since neovim isn't going to interpret or display it correctly.
2. Change neovim to correctly display UnusedCode hints.

In any case, I don't think there's anything pyright can or should do differently here.

**Reply by erictraut** [reply]
Another option is to switch to VS Code with the vim extension.

**Comment by gwww** [comment]
I appreciate your reply. Based on your reply I was to search and find a number of other threads on this subject. I'll step away now ;)
